src/main/java/org/s1ck/gdl/GDLLoader.java

```diff
@@ -338,6 +338,9 @@ public void enterVertex(GDLParser.VertexContext vertexContext) {
     Vertex v;
     if (variable != null && userVertexCache.containsKey(variable)) {
       v = userVertexCache.get(variable);
+      if (!getProperties(vertexContext.properties()).isEmpty()) {
```

Comment:
We should also check if the vertex context has label definitions. Maybe it makes sense to have sth like:
```
private boolean isEmpty(GDLParser.VertexContext vertexContext) {
    return vertexContext.header().label().isEmpty() && vertexContext.properties().property().isEmpty();
}
```

Reply:
I agree, making it clear, that any vertex detail should be set in its first declartion